src/main/scala/chiseltest/backends/SimulatorInterface.scala

```diff
@@ -0,0 +1,30 @@
+package chiseltest.backends
+
+import firrtl.AnnotationSeq
+
+/** API to Simulator. */
+trait SimulatorInterface {
+
+  /** start time of this test. */
+  final val startTime: Long = System.nanoTime()
+
+  /** how many nano seconds has elapsed since this test start. */
+  def elapsedNanoSeconds(): Long = System.nanoTime() - startTime
+
+  def annotations: AnnotationSeq
+
+  /** poke a signal name with a [[BigInt]]. */
+  def poke(signal: String, value: BigInt): Unit
+
+  /** peek a signal name, return a BigInt if found. */
+  def peek(signal: String): Option[BigInt]
```

Comment:
Similarly, perhaps this should throw an exception instead of returning an Option?